### PR TITLE
Updated the code to use ink! `3.0-rc6`

### PR DESCRIPTION
**Pull request by xgreenx**
Updated the code to use ink! `3.0-rc6`.
Added more integration tests to cover cases with the receiver for each token.
Updated `contract` macro to use the type from the trait(it allows the user to not import types used in trait).
`trait_definition` and `contract` macro pass argument to ink! macros.

### CODE REVIEW

**Review by VargSupercolony — APPROVED** [review]
great work :)

File: tests/payment-splitter.tests.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ describe('MY_PAYMENT_SPLITTER', () => {
 
     // Act - Send native token and release them
     await expect(contract.contract.query.totalReleased()).to.have.output(0)
-    await expect(fromSigner(contract.contract, contract.alice.address).tx.receive({ value: 20000000 })).to.eventually.be.fulfilled
+    await expect(contract.contract.tx.receive({ value: 20000000 })).to.eventually.be.fulfilled
     await expect(contract.contract.tx.release(kayne.address)).to.eventually.be.fulfilled
     await expect(contract.contract.tx.release(ian.address)).to.eventually.be.fulfilled
```
Comment (coreggon11) [inline tests/payment-splitter.tests.ts:37]: this test case does not work for me on europa

Comment (coreggon11) [reply tests/payment-splitter.tests.ts:37]: getting `AssertionError: expected promise to be fulfilled but it was rejected with 'Error: createType(Result<Null, CommonErrorsPaymentSplitterPaymentSplitterError>):: DoNotConstruct: Cannot construct unknown type CommonErrorsPaymentSplitterPaymentSplitterError'`

Comment (xgreenx) [reply tests/payment-splitter.tests.ts:37]: Yes, it is problem of europa. I've already reported tp PatractLabs=)

**Review by coreggon11 — APPROVED** [review]

**Pull request merged**